src/action/auth-mobile.js

```diff
@@ -0,0 +1,202 @@
+
+  /**
+   * Initialize the set pin view by resetting input values
+   * and then navigating to the view.
+   * @return {undefined}
+   */
+  initSetPin() {
+    this._store.auth.newPin = '';
+    this._store.auth.pinVerify = '';
+    this._nav.goSetPin();
+  }
+
+  /**
+   * Initialize the pin view by resetting input values
+   * and then navigating to the view.
+   * @return {undefined}
+   */
+  initPin() {
+    this._store.auth.pin = '';
+    this._nav.goPin();
+  }
+
+  /**
+   * Append a digit input to the pin parameter.
+   * @param  {string} options.digit The digit to append to the pin
+   * @param  {string} options.param The pin parameter name
+   * @return {undefined}
+   */
+  pushPinDigit({ digit, param }) {
+    const { auth } = this._store;
+    if (auth[param].length < PIN_LENGTH) {
+      auth[param] += digit;
+    }
+    if (auth[param].length < PIN_LENGTH) {
+      return;
+    }
+    if (param === 'newPin') {
+      this._nav.goSetPinConfirm();
+    } else if (param === 'pinVerify') {
+      this.checkNewPin();
+    } else if (param === 'pin') {
+      this.checkPin();
+    }
+  }
+
+  /**
+   * Remove the last digit from the pin parameter.
+   * @param  {string} options.param The pin parameter name
+   * @return {undefined}
+   */
+  popPinDigit({ param }) {
+    const { auth } = this._store;
+    if (auth[param]) {
+      auth[param] = auth[param].slice(0, -1);
+    } else if (param === 'pinVerify') {
+      this.initSetPin();
+    }
+  }
+
+  /**
+   * Check the PIN that was chosen by the user was entered
+   * correctly twice to make sure that there was no typo.
+   * If everything is ok, store the pin in the keystore and
+   * unlock the wallet.
+   * @return {Promise<undefined>}
+   */
+  async checkNewPin() {
+    const { newPin, pinVerify } = this._store.auth;
+    if (newPin.length !== PIN_LENGTH || newPin !== pinVerify) {
+      this._alert('Incorrect PIN', () => this.initSetPin());
+      return;
+    }
+    await this._setToKeyStore(PIN, newPin);
+    await this._generateWalletPassword();
+  }
+
+  /**
+   * Check the PIN that was entered by the user in the unlock
+   * screen matches the pin stored in the keystore and unlock
+   * the wallet.
+   * @return {Promise<undefined>}
+   */
+  async checkPin() {
+    const { pin } = this._store.auth;
+    const storedPin = await this._getFromKeyStore(PIN);
+    if (pin !== storedPin) {
+      this._alert('Incorrect PIN', () => this.initPin());
+      return;
+    }
+    await this._unlockWallet();
+  }
+
+  //
+  // TouchID & KeyStore Authentication
+  //
+
+  /**
+   * Try authenticating the user using either via TouchID/FaceID on iOS
+   * or a fingerprint reader on Android.
+   * @return {Promise<undefined>}
+   */
+  async tryFingerprint() {
+    const hasHardware = await this._Fingerprint.hasHardwareAsync();
+    if (!hasHardware) {
+      return;
+    }
+    const msg = 'Unlock your Wallet';
+    const { success } = await this._Fingerprint.authenticateAsync(msg);
```

Comment:
Can confirm that fingerprint auth works on Android!!

Reply:
Nice. Did you get that iOS device yet?